docs/cost-optimisation.md

```diff
@@ -0,0 +1,148 @@
+# Cost Optimisation
+
+## Limit The Total Number Of Workflows And Pods
+
+> Suitable for all.
+
+A workflow (and for that matter, any Kuberenetes resource) will incure a cost as long as they exists in your cluster. 
+
+The workflow controller memory and CPU needs increase linearly with the number of pods and workflows you are currently running. 
+
+Limit the total number of workflows using:
+
+* Active Deadline - delete running workflow that do not complete in a set time
+* [Workflow TTL Strategy](fields.md#ttlstrategy) - delete completed workflows after a time
+* [Pod GC](fields.md#podgc) - delete completed pods after a time
+
+Example
+
+```
+spec:
+  # must complete in 8h (28,800 seconds)
+  activeDeadlineSeconds: 28800
+  # keep workflows for 1d (86,400 seconds)
+  ttlStrategy:
+    secondsAfterCompletion: 86400
+  # delete all pods as soon as they complete
+  podGC:
+    strategy: OnPodCompletion
+```
+
+You can set these configurations globally using [Default Workflow Spec](default-workflow-specs.md).
+
+If you need to keep records historically, use the [Workflow Archive](workflow-archive.md).
+
+Changing these settings will not delete workflows that have already run. To list old workflows:
+
+```
+argo list --completed --since 7d
+```
+
+> v2.9 and after
+
+To list/delete workflows completed over 7 days ago:
+
+```
+argo list --older 7d
+argo delete --older 7d
+```
+
+## Set Resources Requests Of Your Argo Instances
+
+> Suitable if you have many instances, e.g. on dozens of clusters or namespaces.
+
+Set a resource quota for the namespace you install Argo in to limit its total usage, e.g.
+
+```
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: argo
+spec:
+  hard:
+    pods: "4"
+    limits.cpu: 1000m
+    limits.memory: 1Gi
+    requests.cpu: 500m
+    requests.memory: 128Mi
+```
+
+Use limit range to set default container requests and limits, e.g.
+
+```
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: argo
+spec:
+  limits:
+    - type: Container
+      defaultRequest:
+        cpu: 100m
+        memory: 64Mi
+      default:
+        cpu: 500m
+        memory: 128Mi
+```
+
+This above limit is suitable for the Argo Server, as this is stateless. The Workflow Controller is stateful and will scale to the number of live workflows - so you are likely to need higher values.
```

Comment:
I don't see how ResourceQuota and LimitRange are useful for sizing argo-server or workflow-controller. It is much more straightforward to set these as CPU requests and limits on those containers themselves, rather than indirectly through ResourceQuota and LimitRange. 

I would either delete this section, or convert it to a tuning guide on setting requests/limits on the argo-server and controller.

Reply:
done